M2/Macaulay2/packages/MonomialIntegerPrograms.m2

```diff
@@ -0,0 +1,1656 @@
+  
+   An excellent user guide to using Zimpl can be found at 
+   @HREF"https://zimpl.zib.de/download/zimpl.pdf"@. The author, Thorsten
+   Koch, requests that research making use of this software please
+   cite his 2004 PhD thesis, {\em Rapid Mathematical Programming}. The
+   appropriate BibTeX entry can be found here: @HREF"https://zimpl.zib.de/download/zimpl.bib"@.
+   Zimpl is distributed under GPL.
+  
+   Additionally, any research that uses SCIP needs a proper citation. See the
+   {\bf How to Cite} tab on their home page.  
+     
+   Finally, because this package relies on temporary files, Windows users must
+   ensure the directory @TT"/tmp"@ exists as stated in the documentation of
+   @TO temporaryFileName@.
+     
+   {\bf Behavior of package on load.}
+   
+   The value of @TO symbol ScipPrintLevel@ determines the verbosity.
+   It is set to @TT"0"@ when the package is loaded.
+   
+   The functions @TO codim@, and @TO degree@ are overwritten for inputs with
+   type @TO MonomialIdeal@. Specifically, @TO loadSCIPCodimAndDegree@ is run
+   when the package is loaded.
+ Caveat
+  These methods are designed and tested for ideals and polynomial rings with a
+  field as the coefficients ring. For other rings, the behavior of these methods
+  is unspecified.
+ SeeAlso
+  codimensionIP
+  degreeIP
+  dimensionIP
+  monomialIdealsWithHilbertFunction
+  topMinimalPrimesIP
+  symbol ScipPrintLevel
+///
+
+doc ///
+ Key
+  symbol ScipPrintLevel
+ Headline
+  adjust how much solving information is displayed in MonomialIntegerPrograms
+ Description
+  Text
+   @TT"ScipPrintLevel"@ is a global symbol defined in Monomial Integer Programs 
+   using @TO exportMutable@. After the package has been loaded, 
+   the user can change the value of ScipPrintLevel at any time, 
+   and the specified behavior will immediately apply to all 
+   methods implemented in the package.
+
+   Meaningful options for ScipPrintLevel are:
+   
+   {\bf 0} return the answers to computations only,  suppressing all other printing  
+   
+   {\bf 1} return the answer, and print to screen the location of the
+   temporary directory which contains all the files related to the computation. By 
+   default this is a subdirectory of @TT"/tmp/"@, see @TO temporaryFileName@.
+
+   {\bf 2} all the above, plus display any error or warning messages 
+   generated by SCIP during the computation, i.e. anything sent by SCIP to stderr.
+   See note below about warning messages.
+
+   {\bf 3} all the above, plus print the problem file generated with 
+   this package, used as the input to SCIP
+      
+   {\bf 4} all the above, plus print the solution file generated 
+   by SCIP after solving the IP
+   
+   {\bf 5} all the above, plus print any other information sent by 
+   SCIP to stdout during the solve, if any
+
+   The default value of ScipPrintLevel is 0. To load the package with a different
+   default value for ScipPrintLevel, imitate the following example.
+  CannedExample
```

Comment:
There's no reason to use canned examples, since you're caching example output and can use M2 to generate and save the example output for the documentation.  (add the example output files to the pull request)

Reply:
The reason for a canned example here is to show how to alter the value of ScipPrintLevel which is 0 at default. If it runs as an actual example, then all other examples with print additional information when they are run.

Reply:
Well, there are lots of other examples that are also canned.  And there is no reason I can see for the print level option to be propagated to other examples, for all the examples in a given documentation page run in a separate Macaulay2 process, and load your package separately.

Reply:
I see. @JayWhite2357 do you have any insight here? I remember there was a reason we changed many of the examples to canned examples after the workshop. Was it just to save time creating example output for longer computations?

Reply:
From what I remember, I changed all the examples that took a long time to run into canned examples. I'm not sure why this one is canned. I seem to recall that there was an example where the output was all funky when it was just a normal example.
My guess is that when this isn't canned, all the stderr/comment/package-loading text here are just removed from the example and it ends up being blank. Then, again this might just have been an accident.

Reply:
If an example takes too long to run, it's better to simplify it so it doesn't take as long.  How about converting all the examples to un-canned examples, and let's see how it goes.

Reply:
@lilysilverstein -- are you following this conversation?

Reply:
The last commits got rid of all the Canned Examples